tools/docker/Dockerfile

```diff
@@ -1,33 +1,15 @@
-# Since we support arm64, take advantage of Docker and Zig's cross
-# compilation, rather than running under qemu.
-# https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/
 FROM --platform=$BUILDPLATFORM ubuntu:22.04 AS build
 
-RUN  apt-get update \
-  && apt-get install -y wget xz-utils git
-
+# NB: We are doing something unconventional here. To ensure that docker images contain the exact
+# same TigreBeetle binary we publish as our release, we don't build inside a container, and merely
+# extract the right pre-exist binary (we pick among for options based on build-time DEBUG and
+# TARGETPLATFORM arguments).
+RUN  apt-get update && apt-get install -y unzip
 WORKDIR /opt/beta-beetle
-
-# Copy install_zig by itself, to try and keep cache hits as high as possible.
-RUN mkdir scripts
-COPY scripts/install_zig.sh ./scripts
-
-ENV PATH="${PATH}:/opt/beta-beetle/zig"
-
-# Allows passing in --build-arg DEBUG=true during builds (needed by Github CI for generating debug image)
+COPY ./dist/tigerbeetle .
 ARG DEBUG=false
-
-# With our multiplatform builds, Docker will only repeat steps after the ARG TARGETPLATFORM.
-# So, explicitly install zig before then, to not need to do it twice.
-RUN ./scripts/install_zig.sh
-
-COPY .git .git
-COPY scripts ./scripts
-COPY src ./src
-COPY build.zig ./build.zig
-
 ARG TARGETPLATFORM
-RUN test ${TARGETPLATFORM} = "linux/arm64" && CPU="baseline+aes+neon" || CPU="x86_64_v3+aes"; CI=true ./scripts/install.sh
+RUN unzip tigerbeetle-$(test ${TARGETPLATFORM} = "linux/arm64" && echo "aarch64" || echo "x86_64")-linux$(test ${DEBUG} = "true" && echo "-debug" || echo "").zip
```

Comment:
Open to suggestions how to make this more readable!

Reply:
I think you can replace the whole Dockerfile with:

```dockerfile
FROM alpine

ARG TARGETARCH
COPY dist/docker/tigerbeetle-${TARGETARCH} /tigerbeetle

ENTRYPOINT ["/tigerbeetle"]
```

If in the `publish_docker` (or a `build_docker`) you unzip the desired eg aarch64 tigerbeetle to `dist/docker/tigerbeetle-arm64`.

(even if you don't want to go down that route entirely, it would still be good to eliminate the multi-stage build since it's no longer needed!)

Reply:
`FROM alpine`

Bit more context on that - the only reason we're not `FROM scratch` is so that if you're debugging or poking around you have a somewhat familiar environment to do that in.

Reply:
@cb22 

>I think you can replace the whole Dockerfile with:

This still needs `ARG DEBUG`, no?